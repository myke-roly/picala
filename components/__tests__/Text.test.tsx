import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {Text} from '../Text';

describe('Text Component', () => {
  it('renders with default props', () => {
    const {toJSON} = render(<Text>Hello World</Text>);
    const text = screen.getByText('Hello World');

    expect(text).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with title variant', () => {
    const {toJSON} = render(<Text variant="title">Title Text</Text>);
    const text = screen.getByText('Title Text');
    expect(text).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with subtitle variant', () => {
    const {toJSON} = render(<Text variant="subtitle">Subtitle Text</Text>);
    const text = screen.getByText('Subtitle Text');
    expect(text).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with different levels', () => {
    const {toJSON} = render(<Text level="lg">Large Text</Text>);
    const text = screen.getByText('Large Text');
    expect(text).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with center alignment', () => {
    const {toJSON} = render(<Text center>Centered Text</Text>);
    const text = screen.getByText('Centered Text');
    expect(text).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with strong styling', () => {
    const {toJSON} = render(<Text strong>Strong Text</Text>);
    const text = screen.getByText('Strong Text');
    expect(text).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with custom color', () => {
    const {toJSON} = render(<Text color="secondary">Colored Text</Text>);
    const text = screen.getByText('Colored Text');
    expect(text).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('combines multiple props', () => {
    const {toJSON} = render(
      <Text variant="title" level="lg" color="primary" center strong>
        Combined Props Text
      </Text>
    );
    const text = screen.getByText('Combined Props Text');
    expect(text).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
