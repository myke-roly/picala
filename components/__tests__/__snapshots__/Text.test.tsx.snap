// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component combines multiple props 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "700",
      },
      {
        "fontSize": 20,
      },
      {
        "color": "#1c434e",
      },
      {
        "fontWeight": "700",
      },
      {
        "textAlign": "center",
      },
      undefined,
    ]
  }
>
  Combined Props Text
</Text>
`;

exports[`Text Component renders with center alignment 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "400",
      },
      {
        "fontSize": 16,
      },
      {
        "color": "#1c434e",
      },
      {},
      {
        "textAlign": "center",
      },
      undefined,
    ]
  }
>
  Centered Text
</Text>
`;

exports[`Text Component renders with custom color 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "400",
      },
      {
        "fontSize": 16,
      },
      {
        "color": "#a58266",
      },
      {},
      undefined,
      undefined,
    ]
  }
>
  Colored Text
</Text>
`;

exports[`Text Component renders with default props 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "400",
      },
      {
        "fontSize": 16,
      },
      {
        "color": "#1c434e",
      },
      {},
      undefined,
      undefined,
    ]
  }
>
  Hello World
</Text>
`;

exports[`Text Component renders with different levels 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "400",
      },
      {
        "fontSize": 20,
      },
      {
        "color": "#1c434e",
      },
      {},
      undefined,
      undefined,
    ]
  }
>
  Large Text
</Text>
`;

exports[`Text Component renders with strong styling 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "400",
      },
      {
        "fontSize": 16,
      },
      {
        "color": "#1c434e",
      },
      {
        "fontWeight": "700",
      },
      undefined,
      undefined,
    ]
  }
>
  Strong Text
</Text>
`;

exports[`Text Component renders with subtitle variant 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "600",
      },
      {
        "fontSize": 16,
      },
      {
        "color": "#1c434e",
      },
      {},
      undefined,
      undefined,
    ]
  }
>
  Subtitle Text
</Text>
`;

exports[`Text Component renders with title variant 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "700",
      },
      {
        "fontSize": 16,
      },
      {
        "color": "#1c434e",
      },
      {},
      undefined,
      undefined,
    ]
  }
>
  Title Text
</Text>
`;
